File: C/ex11/is_prime_solution.c
// Is Prime (Exercise 11) - Solution
// Check if a given number is prime.

#include <stdio.h>
#include <math.h>

int main() {
    int n;
    scanf("%d", &n);
    
    if (n < 2) {
        printf("0\n");
        return 0;
    }
    
    if (n == 2) {
        printf("1\n");
        return 0;
    }
    
    if (n % 2 == 0) {
        printf("0\n");
        return 0;
    }
    
    int limit = (int)sqrt(n);
    for (int i = 3; i <= limit; i += 2) {
        if (n % i == 0) {
            printf("0\n");
            return 0;
        }
    }
    
    printf("1\n");
    return 0;
}

File: C#/ex10/is_prime_solution.cs
// Is Prime (Exercise 10) - Solution
// Check if a given number is prime.

using System;

class Program {
    static void Main() {
        int n = int.Parse(Console.ReadLine()!);
        
        if (n < 2) {
            Console.WriteLine("false");
            return;
        }
        
        if (n == 2) {
            Console.WriteLine("true");
            return;
        }
        
        if (n % 2 == 0) {
            Console.WriteLine("false");
            return;
        }
        
        for (int i = 3; i * i <= n; i += 2) {
            if (n % i == 0) {
                Console.WriteLine("false");
                return;
            }
        }
        
        Console.WriteLine("true");
    }
}
